app/forms/validators.py

```diff
@@ -86,57 +98,58 @@ class NumberRange:
 
     def __init__(
         self,
-        minimum=None,
-        minimum_exclusive=False,
-        maximum=None,
-        maximum_exclusive=False,
-        messages=None,
-        currency=None,
+        minimum: NumType = None,
+        minimum_exclusive: bool = False,
+        maximum: NumType = None,
+        maximum_exclusive: bool = False,
+        messages: dict[str, str] = None,
+        currency: str = None,
     ):
         self.minimum = minimum
         self.maximum = maximum
         self.minimum_exclusive = minimum_exclusive
         self.maximum_exclusive = maximum_exclusive
-        self.messages = {**error_messages, **(messages or {})}
+        self.messages: Mapping[str, str] = {**error_messages, **(messages or {})}
         self.currency = currency
 
-    def __call__(self, form, field):
-        value = field.data
-        error_message = None
+    def __call__(
+        self, form: "QuestionnaireForm", field: DecimalFieldWithSeparator
+    ) -> None:
+        value: Union[int, Decimal] = field.data
         if value is not None:
-            if self.minimum is not None:
-                error_message = self.validate_minimum(value)
-            if error_message is None and self.maximum is not None:
+            error_message = self.validate_minimum(value)
+            if error_message is None:
                 error_message = self.validate_maximum(value)
```

Comment:
This could now be simplified:
```suggestion
            error_message = self.validate_minimum(value) or self.validate_maximum(value)
```